server/src/main.rs

```diff
@@ -1,59 +1,83 @@
 mod app;
+mod db;
 mod graphql;
+mod settings;
 
-use actix_web::{middleware, App, HttpServer};
+#[macro_use]
+extern crate diesel;
+
+use actix_web::{middleware, web, App, HttpServer};
 use anyhow::Result;
-use config::Config;
-use std::path::PathBuf;
-use std::sync::Arc;
+use settings::Settings;
+use std::{
+    convert::TryFrom,
+    net::{IpAddr, SocketAddr},
+    path::PathBuf,
+};
 use structopt::StructOpt;
 
+#[cfg(feature = "autoreload")]
+use listenfd::ListenFd;
+
 #[derive(Debug, StructOpt)]
 #[structopt(name = "stacks_exchange")]
 struct Opt {
     #[structopt(parse(from_os_str), short, long)]
-    conf: Option<PathBuf>
-}
-
-fn init_cfg() -> Result<Config> {
-    let opt = Opt::from_args();
-
-    // always apply default config to ensure all settings defined
-    let mut cfg = Config::new();
-    cfg.merge(config::File::with_name("conf/default.toml")).unwrap();
-
-    // apply a conf override file if one is provided
-    if let Some(path) = opt.conf {
-        cfg.merge(config::File::from(path))?;
-    }
-
-    // apply any conf overrides provided in env (e.g. DEF_addr="...")
-    cfg.merge(config::Environment::with_prefix("DEF"))?;
-
-    pretty_env_logger::init();
-    Ok(cfg)
+    conf: Option<PathBuf>,
 }
 
 #[actix_rt::main]
 async fn main() -> Result<()> {
-    let cfg = init_cfg()?;
+    dotenv::dotenv().ok();
+    let opt = Opt::from_args();
+    let settings = Settings::init(opt.conf)?;
+    pretty_env_logger::init();
```

Comment:
This is a comment for all of us - I think in our MVP requirements, if we want even just our friend group to try the app, we will want something more than logging to STDERR. I'll look into some free solutions for saving traces if nobody has a preference for what we use.

Reply:
I believe that GCP (or whatever cloud provider we choose to use) does retain + provide easy access to stdout/stderr logs, which I think should be sufficient for MVP.

We can augment them with [tracing](https://github.com/tokio-rs/tracing), which can capture traces end-to-end to be consumed by various subscribers. But if we want to use it with Google Cloud's native tracing solution it's probably not as simple as just plugging it in